jux/unit.py

```diff
@@ -134,10 +136,10 @@ def repeat_action(self, success: bool) -> 'Unit':
         pop_only = (act.n <= 1) & ~act.repeat & success & not_empty
         pop_and_push_back = (act.n <= 1) & act.repeat & success & not_empty
 
-        data = jax.tree_map(
+        data: UnitAction = jax.tree_map(
             lambda queue, a: queue.at[self.action_queue.rear].set(a),
             self.action_queue.data,
-            act._replace(n=jnp.where(pop_and_push_back, 1, act.n)),
+            act._replace(n=jnp.where(pop_and_push_back, act.repeat, act.n)),
```

Comment:
new change in v2.1.0 thats more of fixing an old one.

We used to accidentally set `n = 1` when recycling actions to end of the queue., This has been changed to now set `n = repeat` when recycling.